src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c

```diff
@@ -0,0 +1,814 @@
+						xocl_err(&sdm->pdev->dev, "Unable to create sysfs node (%s), err: %d\n", sysfs_name[2], err);
+					}
+				}
+
+				//Create *_avg sysfs node
+				if(strlen(sysfs_name[3]) != 0) {//not empty
+					err = hwmon_sysfs_create(sdm, sysfs_name[3], repo_type, SENSOR_AVG_VAL, srecords);
+					if (err) {
+						xocl_err(&sdm->pdev->dev, "Unable to create sysfs node (%s), err: %d\n", sysfs_name[3], err);
+					}
+				}
+			}
+		}
+
+		remaining_records--;
+		srec++;
+		srecords++;
+	}
+
+	memcpy(repo_object->eor.eor_marker, &in_buf[buf_index], sbyte * 3);
+	buf_index += 3;
+	*(repo_object->eor.eor_marker + 3) = '\0';
+	sdm->sysfs_iter = NULL;
+
+	return repo_object;
+}
+
+static void destroy_hwmon_sysfs(struct platform_device *pdev)
+{
+	struct xocl_hwmon_sdm *sdm;
+
+	sdm = platform_get_drvdata(pdev);
+
+	if (!sdm->supported)
+		return;
+
+	if (sdm->hwmon_dev) {
+		hwmon_device_unregister(sdm->hwmon_dev);
+		sdm->hwmon_dev = NULL;
+	}
+}
+
+static int create_hwmon_sysfs(struct platform_device *pdev)
+{
+	struct xocl_hwmon_sdm *sdm;
+	struct xocl_dev_core *core;
+	int err;
+
+	sdm = platform_get_drvdata(pdev);
+	core = XDEV(xocl_get_xdev(pdev));
+
+	if (!core) {
+		xocl_err(&pdev->dev, "xocl_get_xdev returns NULL...\n");
+		return 0;
+	}
+
+	if (!sdm->supported)
+		return 0;
+
+	sdm->hwmon_dev = hwmon_device_register(&core->pdev->dev);
+	if (IS_ERR(sdm->hwmon_dev)) {
+		err = PTR_ERR(sdm->hwmon_dev);
+		xocl_err(&pdev->dev, "register sdm hwmon failed: 0x%x", err);
+		goto hwmon_reg_failed;
+	}
+
+	dev_set_drvdata(sdm->hwmon_dev, sdm);
+
+	xocl_err(&pdev->dev, "created hwmon sysfs list");
+	sdm->sysfs_created = true;
+
+	return 0;
+
+create_bdinfo_failed:
+	hwmon_device_unregister(sdm->hwmon_dev);
+	sdm->hwmon_dev = NULL;
+hwmon_reg_failed:
+	return err;
+}
+
+static int hwmon_sdm_remove(struct platform_device *pdev)
+{
+	struct xocl_hwmon_sdm *sdm;
+
+	sdm = platform_get_drvdata(pdev);
+	if (!sdm) {
+		xocl_err(&pdev->dev, "driver data is NULL");
+		return -EINVAL;
+	}
+
+	if (sdm->sysfs_created)
+		destroy_hwmon_sysfs(pdev);
+
+	platform_set_drvdata(pdev, NULL);
+
+	return 0;
+}
+
+static ssize_t bdinfo_show(struct device *dev,
+	struct device_attribute *attr, char *buf)
+{
+	struct xocl_hwmon_sdm *sdm = dev_get_drvdata(dev);
+	xdev_handle_t xdev = xocl_get_xdev(sdm->pdev);
+	int i = 0;
+
+	char *in_buf = xocl_xgq_collect_bdinfo_sensors(xdev);
+	struct sdr_response* resp = parse_sdr_info(in_buf, sdm, false);
+	display_sensor_data(sdm, resp);
+
+	return sprintf(buf, "%d\n", i);
+}
+static DEVICE_ATTR_RO(bdinfo);
+
+static ssize_t all_sensors_show(struct device *dev,
+	struct device_attribute *attr, char *buf)
+{
+	struct xocl_hwmon_sdm *sdm = dev_get_drvdata(dev);
+	xdev_handle_t xdev = xocl_get_xdev(sdm->pdev);
+	int i = 0;
+
+	char* in_buf = xocl_xgq_collect_power_sensors(xdev);
+	struct sdr_response* resp = parse_sdr_info(in_buf, sdm, false);
+	display_sensor_data(sdm, resp);
+
+	char* in_buf1 = xocl_xgq_collect_temp_sensors(xdev);
```

Comment:
/proj/xsjhdstaff2/davidzha/XRT_zocl/src/runtime_src/core/pcie/driver/linux/xocl/mgmtpf/../subdev/hwmon_sdm.c:132:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-
statement]
   uint8_t name_type = (srec->name_type_length >> SDR_TYPE_POS) & SDR_TYPE_MASK;

you seem have lots of warnings, please run 
make clean
make